src/coreclr/jit/emitarm64.cpp

```diff
@@ -11816,11 +11820,18 @@ void emitter::emitDispImm(ssize_t imm, bool addComma, bool alwaysHex /* =false *
 
         if ((imm & 0xFFFFFFFF00000000LL) != 0)
         {
-            printf("0x%llx", imm);
+            if (isAddrOffset)
+            {
+                printf("%08XH", imm);
```

Comment:
Is this a correct format specifier for a 64-bit int?

Reply:
The change does not really make sense to me, this immediate always has upper bits set so it is already printing at least 8 characters.

Reply:
not really. It should be still `%llx`. Also, I realized that one of my change (that enables this for arm64) was not pushed. Pushed it now.